### DISCUSSION TITLE
Application embedding / Reusing helpers or database connectors in standalone files

### DISCUSSION THREAD

**Discussion by gregorkrambs** [mojolicious/mojo.js#101] [Q&A]
Hi,

coming from Perl, I'm really interested in the speed and in-built async support of mojo.js. However, my architecture relies on heavily using standalone scripts (e.g. for cronjobs) that are reusing the config files, helpers and database connectors.

The Perl Mojolicious docs are describing this as Application embedding. (https://docs.mojolicious.org/Mojolicious/Guides/Cookbook#Application-embedding).

For mojo.js, it seems there is just an easy way to reuse config files (https://mojojs.org/docs/Cookbook.md#adding-a-configuration-file), but I can't really find anything about reusing other parts of a main app without importing everything again.

Is there a trick to do this? Thanks in advance for any suggestions!

**Comment by kraih** [comment]
Application embedding became a thing in Mojolicious mostly because of tests. That's not really needed  with mojo.js because using ESM exports is [so easy](https://github.com/mojolicious/pg.js/blob/1fe70f66c148252800fb405424736fe9fa607622/examples/blog/test/blog.js#L1). For cronjob scripts i tend to use [application specific commands](https://mojojs.org/docs/Cookbook.md#adding-commands-to-your-application) in both frameworks.

**Reply by gregorkrambs** [reply]
Just tested application specific commands. They work like a charm!

Thanks for the quick help and for the awesome web framework that I've been using and loving for years now.
